src/runtime_src/core/tools/common/JSONConfigurable.h

```diff
@@ -17,46 +17,65 @@
 #ifndef __JSONConfigurable_h_
 #define __JSONConfigurable_h_
 
+#include "XBUtilitiesCore.h"
+
 #include <boost/algorithm/string.hpp>
 #include <boost/program_options.hpp>
 #include <boost/property_tree/ptree.hpp>
 
 class JSONConfigurable {
```

Comment:
I am looking through all this, but I can't seem to find where a JSONConfigurable is made, or why it is managed as shared?  It may very well make sense, but some documentation between line 24 and line 26 would be nice :-)

Reply:
Ill add some comments in. The purpose of JSONConfigurable is to serve as the base class of all items that can be displayed via the help menu and can be represented via the device class option JSON file. It helps when generating the help menus quite a bit!